text/0001-cdk-update.md

```diff
@@ -0,0 +1,285 @@
+mitigate this, `accelerate` only runs on stacks that have explicitly opted in to
+the capability.
+
+### What changes are required to enable this change?
+
+1. An implementation of the `cdk deploy --accelerated` command in the CDK
+   toolkit that can examine the Application stacks for updates that can be
+   applied and apply them.
+2. An expansion of the CDK asset model to include optional construction process
+   metadata and construction input metadata. This is required to allow CDK to
+   identify input changes to the asset, reconstruct it from the defined
+   processes, and then publish it to the necessary AWS service.
+
+### Is this a breaking change?
+
+No, it is not.
+
+### What are the drawbacks of this solution?
+
+- Updating like this still entails a fair amount of preprocesing time, since for
+  complex projects (Golang and Java lambdas, for example) there remains a
+  compilation step.
+- The update model requires the AWS account used to publish the stack to also
+  have sufficient permissions to update the underlying resources, rather than
+  simply requiring the account to have CloudFormation access.
+- Runtimes that require compilation or assembly -- Go or Java lambdas, docker
+  images -- do not benefit from the `--watch` support as naturally, and require
+  some manual steps.
+- updating Lambda functions that are using aliases and provisioned concurrency
+  can take several minutes to switch over between the two versions.
+
+### What alternative solutions did you consider?
+
+We considered SAM accelerate for a similar purpose, but SAM covers a relatively
+small set of possible application structures in AWS, while CDK is intended to
+address the whole set of them.
+
+We also considered introspecting the CDK model ourselves, but concluded that
+there was little value in reinventing the wheel, so to speak, when CDK already
+had all of the information we'd need to deliver this.
+
+For identifying stacks that are subject to accelerated deployment, we are
+considering defining a full "Personal Development Stack" model, possibly based
+off of information in the CDK `App` context.
+
+### What is the high level implementation plan?
+
+We will start from the prototype CDK update command that identifies the Lambda
+resources and then publishes them using the CDK toolkit. We will expand it to
+generalize it to use a simple plugin model to define the update mechanism, which
+will allow for ECS support in addition to Lambda support.
+
+Additionally, a `--watch` flag and a file watcher will be added to support
+monitoring the inputs of assets for changes.
+
+### Are there any open issues that need to be addressed later?
+
+- This RFC requires changes that may affect
+  [rfc-0092 - cdk assets](https://github.com/aws/aws-cdk-rfcs/blob/master/text/0092-asset-publishing.md),
+  as it entails changes to the asset model to support asset transformation and
+  sources.
+- This RFC can be extended to add support for further pluggable asset and update
```

Comment:
What's the design of the extensibility model? What does it take to add a new resource type? Can this be done outside of the CLI codebase or does it require a change to the CLI? (I am okay with both models).

Reply:
I don't actually know; typescript is new enough to me that I'm not sure what affordances the language has for extensibility that way. I _suspect_ that the best design here will be to actually embed the extensions in the constructs themselves, so that we can allow for this in constructs we haven't yet thought of, or even in higher level constructs that are amenable to this approach (think SAM applications, for example). I'll put that down in the RFC and we can discuss it.

Reply:
I don't think this is about typescript. Adding this to constructs sounds appealing but I am not sure how it will work.

I would start with articulating the straightforward solution and evolve from there.